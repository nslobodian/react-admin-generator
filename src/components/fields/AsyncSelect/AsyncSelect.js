import React from 'react'
import pure from 'recompose/pure'
import { Async } from 'react-select'

export const AsyncSelect = pure(
  ({
     input: { value, onChange, onBlur, name }, meta: { touched, invalid, error },
     loadOptions, labelKey, valueKey,
  }) =>
    <div>
      <Async
        value={value}
        multi
        onChange={val => {
          onChange(val)
          onBlur(val)
        }}
        loadOptions={loadOptions}
        labelKey={labelKey}
        valueKey={valueKey}
      />
      {touched && invalid &&
      <div className='alert alert-danger m-t-1' role='alert'>{error}</div>}
    </div>
)

export default AsyncSelect
