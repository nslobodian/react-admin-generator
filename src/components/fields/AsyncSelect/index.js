import AsyncSelect from './AsyncSelect'

export default AsyncSelect
