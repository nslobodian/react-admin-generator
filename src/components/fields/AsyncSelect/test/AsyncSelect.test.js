import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { AsyncSelect } from '../AsyncSelect'

const props = {
      input: {
        value: 1,
        onChange: jest.fn(),
        onBlur: jest.fn(),
        name: 'name',
      },
        meta: {
          touched: 'touched',
          invalid: 'invalid',
          error: 'error',
        },
        loadOptions: jest.fn(),
        labelKey: 'labelKey',
        valueKey: 'valueKey',
    }

describe('(Components) AsyncSelect', () => {
  it('Should render component properly', () => {
    const wrapper = shallow(
      <AsyncSelect
        {...props}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('Should render find and click button', () => {
    const wrapper = shallow(
      <AsyncSelect
        {...props}
      />)
    wrapper.find('Async').first().simulate('change', {target: {value: 'value'}})
  })
})
