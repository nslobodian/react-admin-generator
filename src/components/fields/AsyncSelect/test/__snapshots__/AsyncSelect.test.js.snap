// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Components) AsyncSelect Should render component properly 1`] = `
<div>
  <Async
    autoload={true}
    cache={Object {}}
    ignoreAccents={true}
    ignoreCase={true}
    labelKey="labelKey"
    loadOptions={[Function]}
    loadingPlaceholder="Loading..."
    multi={true}
    onChange={[Function]}
    options={Array []}
    searchPromptText="Type to search"
    value={1}
    valueKey="valueKey"
  />
  <div
    className="alert alert-danger m-t-1"
    role="alert"
  >
    error
  </div>
</div>
`;
